### ISSUE TITLE
DBeaver GUI freezes often

### ISSUE THREAD

**Issue by michal-bartak** [dbeaver/dbeaver#22348]
### Description

Since 23.3.0 (or 23.3.1) DBeaver often freezes. It stops refreshing GUI/reacting on inputs for about 5-10 seconds to recover after a while.

Currently cannot confirm it happens only for networked database instances or for db run locally too. But it's frequent and annoying enough to report it. For sure it never happened before.

### DBeaver Version

CE 23.3.x

### Operating System

Fedora 39

### Database and driver

Postgresql 15

### Steps to reproduce

No steps. It's random. 

### Additional context

N/A

**Comment by philosaur** [comment]
I'm experiencing this too.  Periodic 3-7 second stutters that, in my experience, most often occurs when switching from one query window to another.

DBeaver 23.3.1, Postgres DB over SSH, Windows 10

**Comment by codefossa** [comment]
I am experiencing a similar issue with the flatpak release. Since the upgrade to 23.3.1 it seems the way query syntax is being highlighted is causing an issue, which is only based on visual observation. It appears to freeze and after 5 or so seconds will either unfreeze, or more often crash showing `JVM terminated. Exit code=1` along with the java command.

Flathub commit `652677a3c415190abc13152b9ad009f3be73d98ca4fde2cd2122b87f7195ffb0` (23.3.0) works normally.
Flathub commit `9a3361dca0a99dc276da64d45a5b967ed665f58dc44b7420ba843563e369e1e1` (23.3.1) freezes / crashes.

For others in my situation, a temporary solution is to downgrade to 23.3.0:
```bash
flatpak update --commit 652677a3c415190abc13152b9ad009f3be73d98ca4fde2cd2122b87f7195ffb0 io.dbeaver.DBeaverCommunity
```

**Comment by flambchap** [comment]
Hi folks. Can you try disabling our experimental parser and see if it fixes the issue?
<img width="952" alt="image" src="https://github.com/dbeaver/dbeaver/assets/26031301/d452905c-13de-40f5-b207-2e9ec42e1d94">

**Comment by codefossa** [comment]
I've upgraded to the `23.3.2` flatpak and don't appear to be able to recreate the freeze issue so this may not matter longer.

For the sake of answering your question. Downgrading back to `23.3.1` makes it very easy to immediately replicate the issue. I've disabled the option you've highlighted in your image and it does not disable the new syntax highlighting for me. I tried reopening the application as well which still did not prevent the syntax highlighting or the crashes.

If you would like anything else tested, I would be happy to do so, but since this seems to be resolved in `23.3.2` I think we're good to resolve this.
